### ISSUE TITLE
🐛Bug: reported accessibility issues for Switch

### ISSUE THREAD

**Issue by mcmxcdev** [melt-ui/melt-ui#400]
**Describe the bug**
I am in the process of implementing https://www.melt-ui.com/docs/builders/switch and I noticed the following violations reported by `@storybook/addon-a11y`:

![Screenshot from 2023-08-17 14-12-23](https://github.com/melt-ui/melt-ui/assets/16797721/99fdc77f-3ef3-46d0-8dd5-247f59748a06)

Although our version is slightly modified, this can be reproduced when copying the example from the website.

**Expected behavior**
Expect no a11y violations to occur since melt-ui has an explicitly stated focus on it.

**Comment by TGlide** [comment]
Hey!

You're right, the default example of Melt UI for the Switch was not accessible. I've gone ahead and updated the example by providing an `aria-labelled-by` that makes it accessible.

While Melt does provide accessible defaults, **it is still possible to create unacessible components with Melt.** It is your responsibility to ensure they are accessible, we just help you along the way. In this case, you should provide an `aria-label` or `aria-labelledby` attribute in your element in case it doesn't have inner text.

But we can make this experience better! I've gone ahead and created an issue to expose a label element from form builders, to make this more straightforward when using an external label like in your example: #402. We should also document these scenarios, for sure.

Thank you for raising this issue, I'm going to go ahead and close it, but please let me know in case you have any doubts.

**Comment by mcmxcdev** [comment]
Yeah, I understand that melt-ui only provides the building blocks, but I was sure that `aria-labelled-by` was something that should be provided by melt-ui due to the structure of the elements in the example.

Happy that you updated the example already, we can keep this closed. Thanks!

**Comment by TGlide** [comment]
> Yeah, I understand that melt-ui only provides the building blocks, but I was sure that `aria-labelled-by` was something that should be provided by melt-ui due to the structure of the elements in the example.
> 
> Happy to see that you updated the example already, we can keep this closed. Thanks!

I agree with you that it should tbh, hence the issue that was raised 😄 Sometimes though you don't want an external label, in which cases we should document how to make it accessible!
